pkg/cloud/aws/executor.go

```diff
@@ -61,7 +61,7 @@ func (b BackoffExecutor) Execute(ctx context.Context, f RequestFunction) (interf
 		req.SetContext(ctx)
 		err := req.Send()
 
-		if req.HTTPResponse.StatusCode >= http.StatusInternalServerError && req.HTTPResponse.StatusCode != http.StatusNotImplemented {
+		if err == nil && req.HTTPResponse.StatusCode >= http.StatusInternalServerError && req.HTTPResponse.StatusCode != http.StatusNotImplemented {
 			return nil, &InvalidStatusError{
 				Status: req.HTTPResponse.StatusCode,
 			}
```

Comment:
also `return out, nil` instead of `return out, err` please

Reply:
Not sure why this comment is on this line, but the `return out, err` line needs to stay as it is, otherwise it would swallow an error